packages/expo-screen-orientation/android/src/main/java/expo/modules/screenorientation/ScreenOrientationModule.java

```diff
@@ -0,0 +1,282 @@
+package expo.modules.screenorientation;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.ActivityInfo;
+import android.util.DisplayMetrics;
+import android.view.Surface;
+import android.view.WindowManager;
+
+import org.unimodules.core.ExportedModule;
+import org.unimodules.core.ModuleRegistry;
+import org.unimodules.core.Promise;
+import org.unimodules.core.errors.InvalidArgumentException;
+import org.unimodules.core.interfaces.ActivityProvider;
+import org.unimodules.core.interfaces.ExpoMethod;
+import org.unimodules.core.interfaces.LifecycleEventListener;
+import org.unimodules.core.interfaces.services.UIManager;
+
+public class ScreenOrientationModule extends ExportedModule implements LifecycleEventListener {
+  private ActivityProvider mActivityProvider;
+  private Integer mInitialOrientation = null;
+  static final String ERR_SCREEN_ORIENTATION = "ERR_SCREEN_ORIENTATION";
+
+  public ScreenOrientationModule(Context context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return "ExpoScreenOrientation";
+  }
+
+  @Override
+  public void onCreate(ModuleRegistry moduleRegistry) {
+    mActivityProvider = moduleRegistry.getModule(ActivityProvider.class);
+    moduleRegistry.getModule(UIManager.class).registerLifecycleEventListener(this);
+  }
+
+  @Override
+  public void onHostResume() {
+    Activity activity = mActivityProvider.getCurrentActivity();
+    if (activity != null && mInitialOrientation == null) {
+      mInitialOrientation = activity.getRequestedOrientation();
+    }
+  }
+
+  @Override
+  public void onHostPause() {
+
+  }
+
+  @Override
+  public void onHostDestroy() {
+
+  }
+
+  @Override
+  public void onDestroy() {
+    Activity activity = mActivityProvider.getCurrentActivity();
+    if (activity != null && mInitialOrientation != null) {
+      activity.setRequestedOrientation(mInitialOrientation);
+    }
+  }
+
+  @ExpoMethod
+  public void lockAsync(String orientationLockStr, Promise promise) {
+    Activity activity = mActivityProvider.getCurrentActivity();
+
+    try {
+      OrientationLock orientationLock = OrientationLock.valueOf(orientationLockStr);
+      int orientationAttr = orientationLockJSToNative(orientationLock);
+      activity.setRequestedOrientation(orientationAttr);
+    } catch (IllegalArgumentException e) {
+      promise.reject("ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK", "An invalid OrientationLock was passed in: " + orientationLockStr, e);
```

Comment:
`ERR_SCREEN_ORIENTATION` is predefined as a static member, can you do the same for `ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK`?